develop/interoperability/intro-to-xcm.md

```diff
@@ -1,4 +1,110 @@
+
+A core part of the vision that XCM provides is improving communication between the chains to make system parachains a reality. For example, the Polkadot relay chain handles more than just parachain management and shared security - it handles user balances/assets, governance, and staking. Ideally, the relay chain should be for what it's intended to be - a place for shared security. System parachains can alleviate these core responsibilities from the relay chain but only by using a standard format like XCM.
+
+This is where system parachains come in, where each of these core responsibilities can be delegated to a system parachain respectively.
+
+## XCM Example
+Below shows a simple example of an XCM message. In the example, the native token is withdrawn from the account of Alice and deposited into the account of Bob. The message simulates a transfer between the two accoubts in the same consensus system (ParaA).  You can find the complete example in the [repo](https://github.com/paritytech/xcm-docs/tree/main/examples){target=_blank}.
+
+```rust
+--8<-- 'code/develop/parachain-devs/interoperability/XCM/XCM-first-look.rs'
+```
+The message consists of three instructions
+
+* **WithdrawAsset**
+```rust
+--8<-- 'code/develop/parachain-devs/interoperability/XCM/XCM-withdrawasset.rs'
+```
+    The first instruction takes as an input the MultiAsset that should be withdrawn. The MultiAsset describes the native parachain token with the `Here` keyword. The `amount` parameter is the number of tokens that are transferred. The withdrawal account depends on the origin of the message. In this example the origin of the message is Alice. The `WithdrawAsset` instruction moves `amount` number of native tokens from Alice's account into the holding register.
+
+* **BuyExecution**
+```rust
+--8<-- 'code/develop/parachain-devs/interoperability/XCM/XCM-buyexecution.rs'
+```
+    To execute XCM instructions, weight (some amount of resources) has to be bought. The amount of weight needed to execute an XCM depends on the number and type of instructions in the XCM. The `BuyExecution` instruction pays for the weight using the `fees`. The `fees` parameter describes the asset in the holding register that should be used for paying for the weight. The `weight_limit` parameter defines the maximum amount of fees that can be used for buying weight. There are special occasions where it is not necessary to buy weight. 
+
+* **DepositAsset**
+```rust
+--8<-- 'code/develop/parachain-devs/interoperability/XCM/XCM-depositasset.rs'
+```
+    The `DepositAsset` instruction is used to deposit funds from the holding register into the account of the beneficiary. You dont actually know how much is remaining in the holding register after the `BuyExecution` instruction, but that does not matter since you can specify a wildcard for the asset(s) which should be deposited.
```

Comment:
add reference to:
- DepositAsset - https://github.com/polkadot-fellows/xcm-format?tab=readme-ov-file#depositasset
- BuyExecution - https://github.com/polkadot-fellows/xcm-format?tab=readme-ov-file#buyexecution

Reply:
Done & done, ty!